docs/vcl.md

```diff
@@ -0,0 +1,80 @@
+## Introduction and Overview
+
+VCL is a communication library in VPP that provides a socket-based interface for client/server communication with VPP. It is designed to enable client applications to establish connections with VPP, send and receive control messages, and perform configuration and management operations.
+
+## VCL feature in CalicoVPP
+
+### Enabling VCL feature in CalicoVPP
+
+To enable vcl in your calicoVPP cluster, make sure parameter is set here:
+
+```yaml
+# dedicated configmap for VPP settings
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: calico-vpp-config
+  namespace: calico-vpp-dataplane
+data:
+  CALICOVPP_FEATURE_GATES: |-
+  {
+    "vclEnabled": true
+  }
+```
+
+### Using VCL in pods:
+
+To enable VCL on a pod, use the following annotation:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: samplepod
+  annotations:
+    "cni.projectcalico.org/vppVcl": "enable"
+```
+This will trigger vpp to create the VCL socket and add a session, which is an abstraction that allows applications to send and receive data over the network.
+
+### Sockets
+
+VCL in VPP uses a socketfile: a Unix domain socket used for communication between endpoints. This allows sessions to initiate connections.
+We use an abstract socket for our interface. This socket address is usually represented as an abstract name beginning with a null byte followed by the desired name, typically "@mysocket".
+VPP api has a particular syntax for abstract sockets: using the keyword `abstract:` then `netns_name:`. 
+For example: `abstract:vpp/session,netns_name=/var/run/netns/cni-75e26661-4119-90a4-b9d2-0b00489f76f3`. This syntax appears in vppctl.
+
+## Troubleshooting vcl interface creation
+
+To check abstract socket creation in linux, you can run the following command on the pod running vcl:
+
+```bash
+lsof -U | grep session
+vpp_main 3046023 root   33u  unix 0xffff9eaeb1943740      0t0 396363922 @vpp/session type=SEQPACKET
+```
+
+To check sessions in vpp cli:
```

Comment:
We should probably add a snippet on what sessions are (i.e. listeners or established connections in various protocols). Probably doing a parallel with the output of `ss` and `netstat` would help, before describing `show session verbose`
A warning should probably be added stating that the output might be huge on live systems with a lot of connections